gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/ProductDependency.groovy

```diff
@@ -79,72 +80,86 @@ class ProductDependency implements Serializable {
 
         Preconditions.checkArgument(
                 minimumVersion != maximumVersion,
-                "minimumVersion and maximumVersion must be different in product dependency on %s. This prevents a "
+                "minimumVersion and maximumVersion must be different. This prevents a "
                         + "known antipattern where services declare themselves to require a lockstep upgrade.",
-                productName)
+                SafeArg.of("productGroup", productGroup),
+                SafeArg.of("productName", productName))
 
         if (minimum.isPresent()) {
             Preconditions.checkArgument(
                     maximum.compare(minimum.get()) >= 0,
-                    "Minimum version (%s) is greater than maximum version (%s)",
-                    minimumVersion, maximumVersion)
+                    "Minimum version is greater than maximum version",
+                    SafeArg.of("minimumVersion", minimumVersion),
+                    SafeArg.of("maximumVersion", maximumVersion))
         }
 
+        // Minimum can be unset here if the minimumVersion is a non-orderable SLS version, e.g. "1.0.0.dirty"
         if (recommended.isPresent() && minimum.isPresent()) {
             Preconditions.checkArgument(
                     VersionComparator.INSTANCE.compare(recommended.get(), minimum.get()) >= 0,
-                    "Recommended version (%s) is not greater than minimum version (%s)",
-                    recommendedVersion, minimumVersion)
+                    "Recommended version is not greater than minimum version",
+                    SafeArg.of("recommendedVersion", recommendedVersion),
+                    SafeArg.of("minimumVersion", minimumVersion))
         }
 
         if (recommended.isPresent()) {
             Preconditions.checkArgument(
                     maximum.compare(recommended.get()) >= 0,
-                    "Recommended version (%s) is greater than maximum version (%s)",
-                    recommendedVersion, maximumVersion)
+                    "Recommended version is greater than maximum version",
+                    SafeArg.of("recommendedVersion", recommendedVersion),
+                    SafeArg.of("maximumVersion", maximumVersion))
```

Comment:
I'm not convinced all these SafeArgs are worth it here... this is a gradle plugin so we don't ever ship structured logs anywhere???